MapleServer2/PacketHandlers/Game/GuildHandler.cs

```diff
@@ -63,25 +96,165 @@ private void HandleCreate(GameSession session, PacketReader packet)
 
             if (session.Player.Mesos < 2000)
             {
-                // TODO: Reject packets
+                short NoticeCode = 5121;
+                session.Send(GuildPacket.ErrorNotice(NoticeCode));
                 return;
             }
             else
             {
                 session.Player.Mesos -= 2000;
 
                 session.Send(MesosPacket.UpdateMesos(session));
-                session.Send(GuildPacket.Invite(session.Player, guildName));
+                session.FieldManager.BroadcastPacket(GuildPacket.UpdateGuildTag(session.Player, guildName));
                 session.Send(GuildPacket.Create(guildName));
-                session.Send(GuildPacket.UpdateGuild(session, guildName));
-                session.Send(GuildPacket.Create2(session.Player, guildName));
-                session.Send(GuildPacket.Create3(session.Player, guildName));
+
+                string inviter = ""; // nobody because nobody invited the guild leader
+                byte response = 0; // 0 to not display invite notification
 
                 Guild newGuild = new(guildName, new List<Player> { session.Player });
                 GameServer.GuildManager.AddGuild(newGuild);
 
+                session.Send(GuildPacket.UpdateGuild(session, newGuild));
+                session.Send(GuildPacket.MemberBroadcastJoinNotice(session.Player, inviter, response));
+                session.Send(GuildPacket.MemberJoin(session.Player, guildName));
+            }
+        }
+
+        private void HandleDisband(GameSession session, PacketReader packet)
+        {
+            Guild guild = GameServer.GuildManager.GetGuildByLeader(session.Player);
+            if (guild == null)
+            {
+                return;
+            }
+
+            session.Send(GuildPacket.DisbandConfirm());
+            guild.BroadcastPacketGuild(GuildPacket.MemberNotice(session.Player));
+            GameServer.GuildManager.RemoveGuild(guild);
+        }
+
+        private void HandleInvite(GameSession session, PacketReader packet)
+        {
+            string targetPlayer = packet.ReadUnicodeString();
+
+            Guild guild = GameServer.GuildManager.GetGuildByLeader(session.Player);
+            if (guild == null)
+            {
+                return;
+            }
+
+            Player playerInvited = GameServer.Storage.GetPlayerByName(targetPlayer);
+            if (playerInvited == null)
+            {
+                short NoticeCode = 2563;
+                session.Send(GuildPacket.ErrorNotice(NoticeCode));
+            }
+
+            if (playerInvited.GuildId != 0)
```

Comment:
Can you remove the if-else block and only use :
`if (playerInvited.GuildId != 0) return;`
and
`if (guild.Members.Count >= guild.MaxMembers) return;`